packages/rest/src/sequence.ts

```diff
@@ -116,3 +116,77 @@ export class DefaultSequence implements SequenceHandler {
     }
   }
 }
+
+/**
+ * A sequence implementation using middleware chains
+ */
+export class MiddlewareSequence implements SequenceHandler {
+  /**
+   * Constructor: Injects findRoute, invokeMethod & logError
+   * methods as promises.
+   *
+   * @param invokeMiddleware - invoker for registered middleware in a chain.
+   * To be injected via SequenceActions.INVOKE_MIDDLEWARE.
+   */
+  constructor(
+    @inject(SequenceActions.INVOKE_MIDDLEWARE)
+    readonly invokeMiddleware: InvokeMiddleware,
+    @config()
+    readonly options: InvokeMiddlewareOptions[] = [],
```

Comment:
can you use `.configure() ` to inject middleware options, instead of defaulting it ?

Reply:
The `@config` is here to allow `app.configure` to supply the config value.